source/agora/common/Task.d

```diff
@@ -51,4 +51,26 @@ public class TaskManager
         static import vibe.core.core;
         vibe.core.core.sleep(dur);
     }
+
+    /***************************************************************************
+
+        Run an asynchronous task after the set time in vibe.d's
+        can set it to repeat or run it only once.
+
+        Params:
+            timeout = Determines the minimum amount of time that elapses before
+                the timer fires.
+            callback = If non-null, this delegate will be called when the timer
+                fires
+            periodic = Speficies if the timer fires repeatedly or only once
+
+    ***************************************************************************/
+
+    public void timer (Duration timeout, scope void delegate() dg,
+        bool periodic = true)
```

Comment:
How about we make this an `enum`. `enum Periodic { No, Yes }` so then at the call site we will have:

```D
taskman.timer(10.seconds, &dg, Periodic.No);
```

Because if you have this:


```D
taskman.timer(10.seconds, &dg, false);
```

Then it's not obvious what this means without looking at the documentation.

Reply:
I apply the suggestion.